h2o-core/src/main/java/water/rapids/ast/prims/advmath/AstSpearman.java

```diff
@@ -0,0 +1,125 @@
+package water.rapids.ast.prims.advmath;
+
+import water.DKV;
+import water.MRTask;
+import water.Value;
+import water.fvec.Chunk;
+import water.fvec.Frame;
+import water.fvec.Vec;
+import water.rapids.Env;
+import water.rapids.Merge;
+import water.rapids.Val;
+import water.rapids.ast.AstPrimitive;
+import water.rapids.ast.AstRoot;
+import water.rapids.ast.prims.mungers.AstRankWithinGroupBy;
+import water.rapids.vals.ValNum;
+
+public class AstSpearman extends AstPrimitive<AstSpearman> {
+  @Override
+  public int nargs() {
+    return 1 + 3; // Frame ID and ID of two numerical vectors to calculate SCC on.
+  }
+
+  @Override
+  public String[] args() {
+    return new String[0];
+  }
+
+  @Override
+  public Val apply(Env env, Env.StackHelp stk, AstRoot[] asts) {
+    final Value value = DKV.get(asts[1].str());
```

Comment:
you should do something like 

    asts[1].exec(env)).getFrame()

otherwise you would be working with an unevaluated Frame possibly

Reply:
Done.

Reply:
I accidentally deleted the comment about "evaluation" below.

Reply:
Is this done? The code looks the same to me

Reply:
It was not, now it is:
`final Frame originalUnsortedFrame = stk.track(asts[1].exec(env).getFrame());`